### PR TITLE
Minor adb.py and test cleanup.

### PR DESCRIPTION
**Pull request by mbarbella-chromium**
PTAL. There are still a lot of new tests that need to be added, but wanted to keep the cleanup of the existing stuff in this change.

### CODE REVIEW

File: src/python/tests/test_libs/test_utils.py [hunk 1]
```diff
@@ -173,6 +173,19 @@ def slow(func):
...
 
 
+def android_device_required(func):
+  """Skip Android-specific tests if we cannot run them."""
+  reason = None
+  if not environment.get_value('INTEGRATION'):
+    reason = 'Integration tests are not enabled.'
+  elif environment.platform() != 'LINUX':
+    reason = 'Android device tests can only run on a Linux host.'
+  if not environment.get_value('ANDROID_SERIAL'):
```
Comment (inferno-chromium) [inline src/python/tests/test_libs/test_utils.py]: Should this be elif OR maybe different order ?

File: src/python/tests/core/platforms/android/adb_test.py [hunk 2]
```diff
@@ -130,13 +125,29 @@ def test_read_data_from_file_and_write_data_to_file(self):
...
+    """Ensures that the function works correctly when a device is connected."""
     adb.wait_for_device()
     self.assertEqual(adb.get_device_state(), 'device')
 
-  def test_is_package_installed(self):
-    """Tests is_package_installed."""
+
+@test_utils.android_device_required
+class IsPackageInstalledTest(unittest.TestCase):
+  """Tests for wait_for_device."""
```
Comment (inferno-chromium) [inline src/python/tests/core/platforms/android/adb_test.py]: function description needs fixing.

File: src/python/tests/core/platforms/android/adb_test.py [hunk 3]
```diff
@@ -130,13 +125,29 @@ def test_read_data_from_file_and_write_data_to_file(self):
...
     self.assertFalse(adb.is_package_installed('non.existent.package'))
+
+  def test_partial_package_name_not_installed(self):
+    """Tests that com.google is not recognized as an installed package."""
     self.assertFalse(adb.is_package_installed('com.google'))
-    self.assertTrue(adb.is_package_installed('com.google.android.gms'))
+
+  def test_package_installed(self):
+    """Ensures that gms (which should always be available) is installed."""
+    self.assertTrue(adb.is_package_installed('com.google.android.gms'))
```
Comment (inferno-chromium) [inline src/python/tests/core/platforms/android/adb_test.py]: blank line after this.

File: src/python/tests/core/platforms/android/adb_test.py (hunk 2 above)
Comment (mbarbella-chromium) [reply src/python/tests/core/platforms/android/adb_test.py]: Done

File: src/python/tests/test_libs/test_utils.py (hunk 1 above)
Comment (mbarbella-chromium) [reply src/python/tests/test_libs/test_utils.py]: Not using elif was unintended but this should probably be first as well. Fixed both.

**Pull request merged**